src/Compilers/CSharp/Portable/FlowAnalysis/VariableUsePass.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// VariableUsePass marks all referenced variables in the visited node as used (as a read).
+    /// This pass is used for things like attribute arguments and default parameter expressions,
+    /// which are otherwise not analyzed for use references.
+    /// </summary>
+    internal class VariableUsePass : BoundTreeWalkerWithStackGuard
```

Comment:
> Do you suggest inheriting from BoundTreeWalker (implementing VisitExpressionWithoutStackGuard as a passthrough to Visit), or catching and dealing with the InsufficientExecutionStackException?

After examining how this class is used, I would keep it inheriting from BoundTreeWalkerWithStackGuard. The reason is that it looks like DataFlowPass is taking advantage of stack guard, therefore it would make sense to keep using the guard when DataFlowPass uses VariableUsePass. Though, in this case we probably want VariableUsePass 
 to inherit the current recursion depth from DataFlowPass rather than start from zero. Use sites outside of the DataFlowPass will have to deal with InsufficientExecutionStackException explicitly then.